org.uqbar.project.wollok.typeSystem/src/org/uqbar/project/wollok/typesystem/constraints/variables/TypeVariable.xtend

```diff
@@ -58,28 +61,30 @@ class TypeVariable implements ITypeVariable {
 		new TypeVariable(owner) => [setTypeInfo(new ClosureTypeInfo(parameters, returnType))]
 	}
 
-	def static generic(EObject owner, List<String> typeParameterNames) {
```

Comment:
No need for this, all type variables support genericity.
The most difficult part of this is that creation of "instances" of generic types has to be taken care in other places.